packages/cli/src/modules/mcp/tools/execute-workflow.tool.ts

```diff
@@ -0,0 +1,275 @@
+ * In order to "synchronously" execute the workflow,
+ * it is mapping mcp tool inputs to trigger node pin data and starting execution from there.
+ * LIMITATION: Does not properly support workflows with multiple triggers.
+ */
+export const executeWorkflow = async (
+	user: User,
+	workflowFinderService: WorkflowFinderService,
+	activeExecutions: ActiveExecutions,
+	workflowRunner: WorkflowRunner,
+	workflowId: string,
+	inputs?: z.infer<typeof inputSchema>['inputs'],
+): Promise<ExecuteWorkflowOutput> => {
+	const workflow = await workflowFinderService.findWorkflowForUser(workflowId, user, [
+		'workflow:read',
+	]);
+
+	if (!workflow || workflow.isArchived || !workflow.settings?.availableInMCP) {
+		throw new UserError('Workflow not found');
+	}
+
+	const canExecuteWorkflow = isWorkflowEligibleForMCPAccess(workflow);
+
+	if (!canExecuteWorkflow) {
+		throw new UserError(
+			`Only workflows with the following trigger nodes can be executed: ${Object.values(SUPPORTED_MCP_TRIGGERS).join(', ')}.`,
+		);
+	}
+
+	const triggerNode = findMcpWorkflowStart(workflow.nodes);
+
+	if (!triggerNode) {
+		throw new UserError('No supported trigger node found to start the workflow execution.');
+	}
+
+	const runData: IWorkflowExecutionDataProcess = {
+		executionMode: getExecutionModeForTrigger(triggerNode),
+		workflowData: workflow,
+		userId: user.id,
+	};
+
+	// Set the trigger node as the start node and pin data for it
+	// This will enable us to run the workflow from the trigger node with the provided inputs without waiting for an actual trigger event
+	runData.startNodes = [{ name: triggerNode.name, sourceData: null }];
+	runData.pinData = getPinDataForTrigger(triggerNode, inputs ?? {});
+
+	const executionId = await workflowRunner.run(runData);
+
+	if (!executionId) {
+		return {
+			success: false,
+			executionId: null,
+		};
+	}
+	const data = await activeExecutions.getPostExecutePromise(executionId);
+
+	if (data === undefined) {
+		throw new UnexpectedError('Workflow did not return any data');
+	}
+
+	return {
+		success: data.status !== 'error' && !data.data.resultData?.error,
+		executionId,
+		result: data.data.resultData,
+		error: data.data.resultData?.error,
+	};
+};
+
+/**
+ * Gets the first supported trigger node from the workflow nodes.
+ */
+const findMcpWorkflowStart = (nodes: INode[]): INode | undefined => {
+	return nodes.find((node) => {
+		return !node.disabled && Object.keys(SUPPORTED_MCP_TRIGGERS).includes(node.type);
+	});
+};
+
+/**
+ * Gets the execution mode based on the trigger node type.
+ */
+const getExecutionModeForTrigger = (node: INode): WorkflowExecuteMode => {
+	switch (node.type) {
+		case WEBHOOK_NODE_TYPE:
+			return 'webhook';
+		case CHAT_TRIGGER_NODE_TYPE:
+			return 'chat';
+		case FORM_TRIGGER_NODE_TYPE:
+			return 'trigger';
+		default:
+			return 'trigger';
+	}
+};
+
+/**
+ * Constructs pin data for the trigger node based on provided inputs.
+ */
+const getPinDataForTrigger = (
```

Comment:
For every new trigger we want to support, we have to

- extend the input schema
- add to the allowlist
- add transformation here
- add execution mode mapping at `getExecutionModeForTrigger`

This does not look scalable. Also wondering if the node type definition might be a better location. We also have the node schemas already, which I wonder if we can reuse to avoid having to define a schema here? Also this does not account for node type versions, is not aware to changes to node type schemas, etc. 

I take it this is all temporary until we settle on a strategy?